vllm_v1/sample/logits_processor.py

```diff
@@ -0,0 +1,103 @@
+"""A layer that compute logits from hidden_stats."""
+import inspect
+from typing import Optional
+
+import torch
+import torch.nn as nn
+
+from vllm.distributed import (tensor_model_parallel_all_gather,
+                              tensor_model_parallel_gather)
+from vllm.model_executor.layers.vocab_parallel_embedding import (
+    VocabParallelEmbedding)
+from vllm.platforms import current_platform
+from vllm_v1.sample.metadata import SamplingMetadata
+
+
+class LogitsProcessor(nn.Module):
```

Comment:
BTW why do we need a new logits processor and cannot reuse the the old one? Is it because we have new sampling metadata?

Reply:
Fixed.